packages/sdk/src/api/swap-approval.ts

```diff
@@ -0,0 +1,221 @@
+import z from "zod";
+import {
+  ethereumAddress,
+  bigNumberString,
+  positiveInteger,
+  positiveIntString,
+  positiveFloatString,
+  booleanString,
+  stringOrStringArray,
+} from "./validators.js";
+
+// Request schema
+export const BaseSwapQueryParamsSchema = z.object({
+  amount: positiveIntString,
+  tradeType: z.enum(["minOutput", "exactOutput", "exactInput"]).optional(),
+  inputToken: ethereumAddress,
+  outputToken: ethereumAddress,
+  originChainId: positiveIntString,
+  destinationChainId: positiveIntString,
+  depositor: ethereumAddress,
+  recipient: ethereumAddress.optional(),
+  integratorId: z.string().optional(),
+  refundAddress: ethereumAddress.optional(),
+  refundOnOrigin: booleanString.optional(),
+  // DEPRECATED: Use `slippage` instead
+  slippageTolerance: positiveFloatString(50).optional(), // max. 50% slippage
```

Comment:
We should remove this in the SDK because this is a completely new feature. No need for backwards-compatibility